**1).Write a Python Program to Find the Factorial of a Number?**

factorial of n = n! = n*(n-1)*(n-2)*(n-3).......1 upto getting value 1

factorial of 0 and 1 is 1

factorial does not exists for negative numbers

In [12]:
num = int(input("Enter a number to find factorial : "))
factorial = 1

if num < 0:
    print("factorial does not exist for negative numbers")
elif num == 0:
    print("factorial of %s is 1"%(num))
else:
    for i in range(1,num+1):
        factorial = factorial*i
    print("factorial of {0} is {1}".format(num,factorial))

Enter a number to find factorial : 5
factorial of 5 is 120


**2).Write a Python Program to Display the multiplication Table?**

In [18]:
num = int(input("Enter a number for multiplication table : "))

for i in range(1,11):
    print("%s x %s = %s"%(num,i, num*i))

Enter a number for multiplication table : 17
17 x 1 = 17
17 x 2 = 34
17 x 3 = 51
17 x 4 = 68
17 x 5 = 85
17 x 6 = 102
17 x 7 = 119
17 x 8 = 136
17 x 9 = 153
17 x 10 = 170


**3).Write a Python Program to Print the Fibonacci sequence?**

In [45]:
seq_terms = int(input("Enter a numbers of sequence : "))
num1 = 0
num2 = 1
count = 2
if seq_terms <= 0:
    print("please enter a positive number")
elif seq_terms == 1:
    print("fabonacci sequence: %s"%(num1))
else:
    print("fabonacci sequence : ")
    print(num1,",",num2,end=",")
    while count < seq_terms:
        nth = num1 + num2
        print(nth,end=",")
        num1 = num2
        num2 = nth
        count += 1

Enter a numbers of sequence : 10
fabonacci sequence : 
0 , 1,1,2,3,5,8,13,21,34,

**4).Write a Python Program to Check Armstrong Number?**

Armstrong number:

A number is called Armstrong number if it is equal to the sum of the cubes of its own digits.

For example: 153 is an Armstrong number since 153 = 1 * 1 * 1 + 5 * 5 * 5 + 3 * 3 * 3.
(or) 1 ** 3 + 5 ** 3 + 3 ** 3

In [20]:
num = int(input("Enter a number: "))  
summ = 0  
temp = num  
  
while temp > 0:  
    digit = temp % 10  
    summ += (digit ** 3)  
    temp //= 10  
if num == summ:
    print("%s is a armstrong number"%(num))
else:
    print("%s is not a armstrong number"%(num))

Enter a number: 407
407 is a armstrong number


**5).Write a Python Program to Find Armstrong Number in an Interval?**

In [34]:
num1 = int(input("Enter lower range: "))  
num2 = int(input("Enter upper range: "))

for num in range(num1,num2 + 1):
    summ = 0
    temp = num
    while temp > 0:
        digit = temp % 10
        summ += digit ** 3
        temp //= 10
        if num == summ:
            print(num)

Enter lower range: 100
Enter upper range: 500
125
153
216
370
371
407


**6).Write a Python Program to Find the Sum of Natural Numbers?**

In [37]:
num = int(input("Enter a number: "))  
  
if num < 0:  
    print("Enter a positive number")  
else:  
    sum = 0  
   # use while loop to iterate un till zero  
    while(num > 0):  
        sum += num  
        num -= 1  
    print("The sum is",sum)  

Enter a number: 5
The sum is 15
